File: TradeItIosEmsApi/TradeItOAuthLoginPopupUrlForTokenUpdateRequest.m
#import "TradeItOAuthLoginPopupUrlForTokenUpdateRequest.h"

@implementation TradeItOAuthLoginPopupUrlForTokenUpdateRequest

- (id)initWithApiKey:(NSString *)apiKey
              broker:(NSString *)broker
              userId:(NSString *)userId
           userToken:(NSString *)userToken
interAppAddressCallback:(NSString *)interAppAddressCallback {
    self = [super init];

    if (self) {
        self.apiKey = apiKey;
        self.broker = broker;
        self.userId = userId;
        self.userToken = userToken;
        self.interAppAddressCallback = interAppAddressCallback;
    }

    return self;
}

@end

File: TradeItIosEmsApi/TradeItOAuthLoginPopupUrlForTokenUpdateRequest.swift
class TradeItOAuthLoginPopupUrlForTokenUpdateRequest: TradeItRequest {
    
    var apiKey: String
    var broker: String
    var userId: String
    var userToken: String
    var interAppAddressCallback: String
    
    init(
        apiKey: String,
        broker: String,
        userId: String,
        userToken: String,
        interAppAddressCallback: String
    ) {
        self.apiKey = apiKey
        self.broker = broker
        self.userId = userId
        self.userToken = userToken
        self.interAppAddressCallback = interAppAddressCallback
    }
    
}
